# Introduction
Telomeres are repetitive sequences bound by proteins at the ends of chromosomes. They shorten with each round of replication due to the ‚Äúend replication problem,‚Äù the phenomenon in which Okazaki fragments cannot be placed at the very end of a replicating chromosome. This telomere shortening continues until the reach a critically short length. The shortening of the telomere to this critical point results in it appearing to the cell as a one-sided double strand break. The DNA damage response is then activated, usually resulting in cellular senescence. A subset of cells, however, will circumvent cell death by lengthening their telomeres. This often happens via the reactivation of telomerase. However, telomeres can also be lengthened via a mechanism known as alternative lengthening of telomeres (ALT) in which a critically short telomere uses the end of another chromosome as a template for synthesis. Cells that are able to bypass cell death are then referred to as ‚Äúsurvivors‚Äù. Previous work on ALT has been largely qualitative, with a gene being determined to have a role in ALT if its deletion completely eliminated ALT or altered the distribution of survivor types (which are determined by chromosome end structure and are in equal distributions in tlc1Œî). Until recently researchers have not had a method with which to determine the frequency of ALT, preventing detection of ALT supressors (which would increase ALT events), genes that have a role in but are not essential for ALT, or genes that have a role in ALT but do not impact survivor type. In this paper, a method for determining ALT frequency was developed. Yeast that are deficient for telomerase due to a deletion of the RNA component TLC1 are passaged in liquid until they reach senescence. These senescent cells are then plated, and the formation of colonies (survivors) is monitored for 5 days. As TLC1 deletion eliminates telomerase function, any colonies that are able to bypass senescence are presumed to have done so via ALT, and the number of survivors that form can be compared to the number of cells plated to determine ALT frequency. Once overall frequency was determined by deleting a component of telomerase known as TLC1, genes suspected of being involved in ALT were deleted and changes in ALT frequency determined. This method is valuable as researchers are now able to quantify the effect of a genes that suppress or are involved in, but not essential for, ALT.

The figure I want to recreate from Kockler et al 2022 is figure 1c, which shows the frequency of ALT in the context of various gene deletions. This figure shows the overall frequency of ALT to be 2x10-5. These frequencies were determined by deleting TLC1, the RNA component of telomerase. As the telomerase complex is nonfunctional without TLC1, passaging yeast to senescence in the absence of TLC1 therefore allows for the determination of ALT frequency. By deleting TLC1 alone, the overall frequency of ALT was determined to be 2x10^-5. Other genes believed to be involved in ALT were also deleted, and changes in frequency determined in these genetic backgrounds. ALT is believed to occur via a break-induced replication-like method, in which a chromosome end searches the genome for homology and then invades into/ anneals with that homologous region, genes involved in homology search, invasion and annealing were tested. It is also believed that cell cycle arrest is required for cells to form aLT survivors. Unsurprisingly, loss of RAD52, the major annealing protein, completely eliminated ALT, while loss of another RAD59, another annealing protein, and the invasion gene RAD51 and its accessory protein RAD55, and the checkpoint genes RAD9 and RAD24 all resulted in decreases in ALT frequency. Cell cycle arrest is needed for ALT precursors to have sufficient time to become survivors, and so it is unsurprising that loss of checkpoint proteins RAD9 and RAD24 all decrease ALT frequency. In order for chromosome ends of healthy cells to avoid being detected as double strand breaks the telomeres are bound by protein complexes. When binding of these proteins is eliminated, either by gene deletion or telomere shortening, the chromosome ends appear to the cells as DSBs, triggering the DNA damage response and cell cycle arrest. As uncapping the telomeres makes them more prone to being detected as a break in need of repair, it is possible that loss of telomere capping could increase the frequency with which ALT occurs. This is supported by the deletion of the telomere capping protein RIF1 and the loss of RIF1 and RIF2 together, but not the loss of capping protein RIF2 alone. 
# Data
data_for-comp_sci.csv' and 'data_for_comp_sci_edited.csv' contain the same columns, but all the columns survivors and cells_plated are treated as numbers in the edited version, versus characters in the unedited version. Each row is an individual culture. Strain is the strain from which that culture was derived. Genotype is the genotype of that culture. Survivors is the survivors formed on plates from that colony and cells_plated`is the number of cells plated for that colony with extra doublings taken into account.The strains are 6169 (TLC1/tlc1::BSD, MSH2/msh2::kan), 6212(TLC1/tlc1::BSD, MPH1/mph1::kan) and 6205(TLC1/tlc1::BSD, MLH1/mlh1::kan). Experiments were run between 2020 and early 2023.
# Analysis
In this figure, the strain AM3692(TLC1/tlc1::BSD) and its derivatives are used. AM3692 is heterozygous for a deletion of the telomerase component TLC1. Tetrad dissection, in which haploid spores from a single mother cell are separated, is used to isolate haploid tlc1Œî spores with/without the deletion of a second gene of interest. These haploid tlc1Œî colonies are then started in 1 ml YEPD (yeast extract-peptone-dextrose) overnight, after which they are passaged every 24 hours in 4 ml YEPD until senescence is reached. Senescence is reached after 2-3 passages depending on the growth pattern of the strain, with cells being plated after passage 1 or 2, respectively. Cultures with RAD52, RAD51 and SRS2 deficient strains require plating after the first passage, all others can be plated after passage 2. The number of cell/ml passaged is determined by the growth pattern of the strain, with cultures that divide poorly having higher numbers of cells passaged in order to allow enough cells to be plated to detect the formation of survivors. In all cases, cells passaged is kept low enough that only one survivor is expected to form per culture. This allows for the researcher to be confident that survivors are independent from one another. The total number of survivors for a genotype is monitored and then divided by the total adjusted cells plated to determine a frequency for ALT. Rather than recreating this figure, I am interested in the effect of deleting mismatch repair genes such as MSH2 and MLH1. In the main branch of the github repository biol-4386-course-project-Kendra-Musmaker/Data at main ¬∑ Intro-Sci-Comp-UIowa/biol-4386-course-project-Kendra-Musmaker (github.com)is a file data_for_comp_sci. Each row in this file is an individual culture. Column 1A is strain number from which the culture was derived, column B is the genotype of the culture. Column C is the survivors counted for that colony. Column D I the adjusted cells plated for that colony, where adjusted cells plated is calculated as the cells plated multiplied by 2(remaining population doublings). This file contains cultures from strain 6169 (TLC1/tlc1::BSD MSH2/msh2::kan), 6205 TLC1/tlc1::BSD MLH1/mlh1::kan) and AM6212 (TLC1/tlc1::BSD MPH1/mph1::kan). These cultures were obtained from experiments run between 2020 and early 2023. The data was read into python as a csv. The columns survivors and cells_plated were identified set to have the value type float. A new column called individual frequency was added to the dataframe and was determined as survivors/cells_plated for each culture. A dataframe called df_summary was created that contained total survivors (total_survivors), total cells plated (total_cells), frequency, lower and upper CI and error for each genotype. Total_survivors is the sum of the survivors for all cultures of a particular genotype, while total_cells is the sum of cells plated for all cultures of a genotype. Total_frequency is calculated as total survivors/total cells plated, while lower_CI and upper_CI are calculated as f¬±1.96‚àö((f(1-f)/n)), respectively, with f=total_frequency and n=total cells. Adjusted _total frequencies, adj_lower_ci and adj_upper_ci are the previous values multiplied by 10^10 and are used for graphing purposes.
![Figure 1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276521000903-gr1.jpg)
Adjusted_total_frequency was then plotted using a seaborn barplot with genotypes along the x-axis and frequency (x10^10) as the y axis. Error bars were set to equal ‚Äòerror‚Äô from df_summary, with ‚Äòerror‚Äô being adj_upper_ci ‚Äì adj_lower_ci.

Indidvidual data points can be overlayed by overlaying a seaborn scatterplot in which X=genotype and Y=individual frequency (x10^10).
# Results
Figure 2: Frequency of ALT determined by plating senescent cells. Frequency determined as survivors divided by cells plated. Error bars represent 95% confidence interval calculated as as ùëì¬±1.96‚àö((ùëì(1‚àíùëì)ùëõ)), with f = total_frequency and n=total cells. Non-overlapping CI‚Äôs considered significantly different frequencies. Values on the y-axis are x10^6.

Figure2, altfreq
![Figure 2, The Frequency of ALT determined by plating senescent cells](https://github.com/Intro-Sci-Comp-UIowa/biol-4386-course-project-Kendra-Musmaker/blob/main/Analysis/plot_telomer.png)
Frequency for tlc1Œî was found to be 4.4x10^-5 (95% CI 4.2-4.6). This frequency is significantly lower than either mismatch repair gene (MSH2: 1.68x10^-4, CI 1.56-1.81x10^-4; MLH1: 7.4x10^-5, 6.6-8.2x10^-5) or the frequency of the MPH1 deletion (7.9x10^-5, 6.5x10^-5-9.3x10^-5). The MSH2 deletion has the greatest increase in frequency compared to tlc1Œî, with the frequency of the TLC1 deletion being greater than than either of the other mutants, including the mismatch repair gene MLH1.
# Discussion
All genes in this study are observed to cause an increase in survivor formation when deleted. This suggests that all of these genes suppress the formation of survivors when present in the cell. That MPH1 causes an increase in frequency when deleted is unsurprising, as MPH1 is a helicase that is known to disrupt d-loops in break-induced replication (BIR). ALT is thought to occur by a break-induced replication like process in which a short telomeric end invades into another telomere, creating a displacement loop (d-loop). As this d-loop is needed for invasion and subsequent beginning of synthesis, if d-loops are prematurely disrupted by MPH1 in ALT, it follows that there would be less formation of survivors with MPH1 present.
The increase in frequency when MSH2 is deleted is consistent with the occurrence of heteroduplex rejection occurring at early stages of ALT. Heteroduplex rejection occurs when a mismatch is detected between two DNA strands by the mismatch repair machinery and is unwound rather than repaired. It is thought that mismatches may frequently occur in early stages of survivor formation in yeast, as in S. cerevisiae the telomeres consist of degenerate TG1-3 repeats. When these mismatches are identified by MSH2 in complex with either MSH3 or MSH6, this may lead downstream to the unwinding of this ALT precursor. With MSH2 deleted, these precursor at regions of imperfect homology may go undetected and unrepaired and proceed to form ALT survivors.
That MLH1 deletion increases ALT frequency is somewhat more surprising. MLH1 acts downstream of ALT in classical repair as an endonuclease that cleaves DNA, which is not a stage in heteroduplex rejection. There has been uncertainty regarding whether or not MLH1 plays a role in heteroduplex rejection. The findings here, however, suggest that MLH1 may be involved in heteroduplex rejection to inhibit the formation of ALT precursors.
In order to accurately determine if a gene is involved in ALT and what its effect on ALT may be (increase, decrease, to what extent) it is important that there is consistency at the point of data collection. Some cells that are plated are not yet fully senescent and so may form very small colonies that are not survivors, but are visible to the naked eye.  It is important for reproducibility to be consistent in determining what is considered a survivor. Related to this, it is importnant to have consistency in determining which plates to include in analysis. Our system is meant to follow a poisson distribution in which events should be rare (events being survivors). As such the ideal is to have only one survivor per plate. As ALT is stochastic, however, this is often not the case. For results to be reproducible there must be standards in determining a threshold at which we consider the number of survivors on a plate to be inconsistent with independent events.

As far as portraying the data, it is important to accurately portray frequencies relative to their confidence intervals and the number of samples involved in the frequency. With small sample sizes a gene could appear to have an increase over wild-type but have a confidence interval that extends down to zero. It is important to represent the data with its confidence interval and with the spread of the data.

# Reflection
This project was difficult determining how to begin. Once I understood how to define variables such that I didn't have to have a separate one for each genotype it became easier to make a table of summary statistics. The bar plot was not challenging, but it was difficult to determine how to set the correct error bars. However, I learned a lot and understand how python works more than I did previously.

#References
Kockler ZW, Comeron JM, Malkova A. A unified alternative telomere-lengthening pathway in yeast survivor cells. Mol Cell. 2021 Apr 15;81(8):1816-1829.e5. doi: 10.1016/j.molcel.2021.02.004. Epub 2021 Feb 26. PMID: 33639094; PMCID: PMC8052312.
